GVRf/Framework/framework/src/main/jni/engine/renderer/renderer.cpp

```diff
@@ -208,58 +206,10 @@ void Renderer::cullFromCamera(Scene *scene, Camera* camera,
     occlusion_cull(scene, scene_objects, shader_manager, vp_matrix);
 }
 
-/**
- * Set the render states for render data
- */
-void Renderer::setRenderStates(RenderData* render_data, RenderState& rstate) {
-
-    if (!(rstate.render_mask & render_data->render_mask()))
-        return;
-
-    if (render_data->offset()) {
-        GL(glEnable (GL_POLYGON_OFFSET_FILL));
-        GL(
-                glPolygonOffset(render_data->offset_factor(),
-                        render_data->offset_units()));
-    }
-    if (!render_data->depth_test()) {
-        GL(glDisable (GL_DEPTH_TEST));
-    }
-    if (!render_data->alpha_blend()) {
-        GL(glDisable (GL_BLEND));
-    }
-    if (render_data->alpha_to_coverage()) {
-        GL(glEnable(GL_SAMPLE_ALPHA_TO_COVERAGE));
-        GL(
-                glSampleCoverage(render_data->sample_coverage(),
-                        render_data->invert_coverage_mask()));
-    }
-}
-/**
- * Restore the render states for render data
- */
-void Renderer::restoreRenderStates(RenderData* render_data) {
-    if (render_data->cull_face() != RenderData::CullBack) {
-        GL(glEnable (GL_CULL_FACE));
-        GL(glCullFace (GL_BACK));
-    }
-    if (render_data->offset()) {
-        GL(glDisable (GL_POLYGON_OFFSET_FILL));
-    }
-    if (!render_data->depth_test()) {
-        GL(glEnable (GL_DEPTH_TEST));
-    }
-    if (!render_data->alpha_blend()) {
-        GL(glEnable (GL_BLEND));
-    }
-    if (render_data->alpha_to_coverage()) {
-        GL(glDisable(GL_SAMPLE_ALPHA_TO_COVERAGE));
-    }
-}
 
 void Renderer::renderRenderDataVector(RenderState &rstate) {
 
-    if (!do_batching) {
+    if (!do_batching || (dynamic_cast<GLRenderer*>(renderer) == nullptr) ) {
```

Comment:
No need for dynamic cast and enabling rtti just for this. You could have the Renderer interface define a method that lets implementations tell you do they support batching or not.